### ISSUE TITLE
Move PHP IntelliSense out of core

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/vscode#16586]
https://github.com/felixfbecker/php-language-server has recently gained advanced completion support which goes way beyond VS Code's internal PHP completion. It resolves types, looks at docblocks and suggests everything from classes, properties, methods etc. depending on context. Validation is also implemented.

It is currently blocked by https://github.com/Microsoft/vscode-languageserver-node/issues/127, but after that I plan to release it. The problem though is that there is no way to disable the built-in completion and snippets for PHP, so you will get a lot of duplicated results (because the built-in completion is not context-aware).

Since PHP is the only language that has bundled completion and validation, I would propose to just remove it from core (only keep syntax highlighting).

@mousetraps

**Comment by aeschli** [comment]
+1 for that!
@mousetraps Let me know if you want to do this..

**Comment by mousetraps** [comment]
@aeschli I'm in! 

That said, while it is true that this is the only non-fundamental-to-building-vscode language that provides completions out of the box, I believe it would also be the first time we would pull something out of core that has such a significant level of engagement.

So before we make it happen, I think we should try to define "metrics for success" to understand if there's anything more we should measure to understand the resulting the impact on things like new-user experience and user retention. VSCode has struck a chord with a ton of PHP devs, which is awesome, and it would be unfortunate if this change resulted in funneling a greater % of those users to extensions (good) at the expense of the absolute number of users (bad).

I don't have a good answer to this yet, but I'll think on it. Thoughts?

cc @chrmarti

**Comment by felixfbecker** [comment]
@mousetraps If you need more time to think - can we then get some way to disable the built-in completion and snippets for the short term? We can already disable the validation, but not the completion.

**Comment by chrmarti** [comment]
@mousetraps I don't have good answer to the question about the possible impact on new users either. The architectural argument might weigh more anyway. We will also keep working on the discoverability and usability of our extension story (e.g., #16168).

**Comment by aeschli** [comment]
We clearly have to document that in the release notes. Also we must make sure that when a PHP document is opened, there will be the recommendation popup that will recommend to install php-language-server. If that's not enough we can add an extra dialog that explains the situation when one opens a PHP document.
If there are many complaints we can also think of offerering the old support as an extension.
IMO our current support is too simple and blocks the way of a more serious implementation. I'm in favour let the community take over on PHP.

**Comment by bmewburn** [comment]
Here's another php offering that ticks a few boxes
https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client

**Comment by roblourens** [comment]
We now have options to disable the builtin support. I think we should keep the builtin support though.

**Comment by felixfbecker** [comment]
@roblourens Would be curious to hear why?

**Comment by roblourens** [comment]
I think it has some value, I don't think it's getting in anyone's way now that it can be disabled, I think it's nice to have some basic things work out of the box, and it's not taking any effort to maintain, so I don't see why we should remove it.
